src/lib/utils/entry-point.ts

```diff
@@ -384,7 +391,23 @@ function getEntryPointsForPackages(
         }
 
         results.push({
-            displayName: packageJson["name"] as string,
+            displayName:
+                typedocPackageConfig?.displayName ??
+                // if displayName is not configured, use the package name (and version, if configured)
+                `${packageJson["name"]}${
```

Comment:
I have a few concerns here:

1. Including the version in reflection names means that cross-module links must specify the version in the link... resulting in broken links when bumping a version (`foo@1.2.3!bar` vs `foo!bar`)
2. `displayName` gives another way to rename exports, which is already possible through [`@module`](https://typedoc.org/tags/module/)... is there an advantage to this?
3. We should probably warn if trying to add a version and there's no version.
4. The formatting is inconsistent with how `PackagePlugin` sets the name if there is a version. I'd like these to be the same.
5. Even if `displayName` is set, if `includeVersion` is specified, we should probably add the version.

Reply:
1. I have not thought about this case. I will fix it so that we don't end up changing the reflection name, but just the rendering name - I will try.
2. I didn't know about this and I just tried and I found a limitation with this. That is the `@module` comment must be the first block comment above any code. Unfortunately, we can't make use of this since we have our copyright header. See this example: https://github.com/finos/legend-studio/blob/master/packages/legend-art/src/index.ts
3. Will do
4. Will do
5. Will do

Reply:
@Gerrit0 I think to carry around information on `version` I will need to poke into reflection model once again, I'll add a field for `version` there. Are you okay with this? Or is there some other way that I'm not familiar with. Also, I'm trying to do a fair bit in this one PR. My main concern is to allow specifying the `README` per child packages. So if this change for `displayName` and `version` pose to be more complicated, I think I can leave them out of this PR altogether.

Reply:
2 - Makes sense! It stays.
version - Yep, again I don't love it, but it's a reasonable solution.